src/pages/Dashboard/chartSelectors.js

```diff
@@ -71,52 +81,167 @@ export function getExperimentsCount(stats) {
+    ([
+      { date, total: totalSamplesUnprocessed },
+      { total: totalSamplesProcessed }
+    ]) => ({
+      date: date,
+      unprocessed: totalSamplesUnprocessed,
+      processed: totalSamplesProcessed
     })
   );
+}
 
-  return accumulateByKeys(result, ['samples', 'experiments']);
+export function getJobsByStatusOverTime(jobsTimeline, range) {
+  return transformTimeline(jobsTimeline, range, [
+    'pending',
+    'open',
+    'completed',
+    'failed'
+  ]);
 }
 
-export function getJobsByStatusOverTime(stats, jobName) {
-  const result = stats[jobName].timeline.map(dataPoint => ({
-    date: moment.utc(dataPoint.end).format('lll'),
-    total: dataPoint['total'],
-    ...JOB_STATUS.reduce((accum, status) => {
-      accum[status] = dataPoint[status];
-      return accum;
-    }, {})
+/**
+ * Returns a timeline that is sorted chronologically, and that has all datapoints
+ * for a given `range`.
+ * The `/stats` endpoint only returns the datapoints that have some value.
+ * @param {*} timeline as returned by the /stats endpoint
+ * @param {*} range range param that the graph will be displaying
+ */
+function transformTimeline(timeline, range, fields = ['total']) {
+  const result = [...getTimeline(range)].map(date => {
+    const result = {
+      date
+    };
+    for (let field of fields) {
+      result[field] = 0;
+    }
+    return result;
+  });
+
+  for (let observation of timeline) {
+    // find the closest datapoint to `data.start`
+    let closestTemp = null;
+    for (let graphPoint of result) {
+      if (
+        !closestTemp ||
+        Math.abs(moment(observation.start).diff(graphPoint.date)) <
+          Math.abs(moment(observation.start).diff(closestTemp.date))
+      ) {
+        closestTemp = graphPoint;
+      }
+    }
+    if (closestTemp) {
+      // add the total samples to that datapoint
+      for (let field of fields) {
+        closestTemp[field] = observation[field];
+      }
+    }
+  }
+
+  return result.map(({ date, ...rest }) => ({
+    ...rest,
+    date: formatTimeLabel(date, range)
   }));
+}
 
-  return accumulateByKeys(result, ['total', ...JOB_STATUS]);
+/**
+ * Given a range returns the
+ * @param {*} range day/ | week | month | year
+ */
+function* getTimeline(range) {
+  const now = moment().startOf('day');
+
+  const data = {
+    day: {
+      start: now.clone().subtract(1, 'days'),
+      interval: moment.duration(1, 'hour')
+    },
+    week: {
+      start: now.clone().subtract(1, 'weeks'),
+      interval: moment.duration(1, 'days')
+    },
+    month: {
+      start: now.clone().subtract(30, 'days'),
+      interval: moment.duration(1, 'days')
+    },
+    year: {
+      start: now
+        .clone()
+        .subtract(365, 'days')
+        .startOf('month'),
+      interval: moment.duration(1, 'months')
+    }
+  };
+  let nextDate = data[range].start;
+  let interval = data[range].interval;
+  yield nextDate.format();
+
+  while (true) {
```

Comment:
I feel like there's a way to do this with a map

Reply:
Yes, now that you mention it, seems to be possible. If it's ok with you, let's address it in a future refactor.